### DISCUSSION TITLE
Return Stripe checkout URL from checkout method

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#49801] [Q&A]
### Cashier Stripe Version

15.2.0

### Laravel Version

10.41.0

### PHP Version

8.2.*

### Database Driver & Version

MySQL

### Description

I'd like to migrate my subscription checkout process over to the Stripe checkout for better maintainability. My Laravel project serves as an API for a Nuxt JS front-end, this means I need to return the URL of the Stripe checkout to redirect the user to in my axios response in my front-end.

The `checkout` method appears to do something with the redirect method as I see a 303 response code in my browser. I've tried attaching the `toResponse` method to just get the URL with no luck.

I just need some guidance in returning a URL from the checkout method please here:

### Steps To Reproduce

```php
$subscription->create($request->input('token'), [], $subscriptionOptions);

return $subscription->checkout([
    'success_url' => 'http://localhost:3005/account/subscriptions/',
    'cancel_url' => 'http://localhost:3005/account/subscriptions/',
])->toResponse($request);
```

**Comment by rubanraj7** [comment]
I think you may be missing a ` ->newSubscription('default', 'price_monthly')` Try this:

```
return $subscription->newSubscription('default', 'price_monthly')->checkout([
    'success_url' => 'http://localhost:3005/account/subscriptions/',
    'cancel_url' => 'http://localhost:3005/account/subscriptions/',
])->toResponse($request); 
```


https://laravel.com/docs/10.x/billing#subscription-checkouts

**Comment by stuartcusackie** [comment]
A `toArray()` function is available on the Checkout class. It contains a url to stripe checkout.

I haven't used it yet but I'll find out how useful it is soon.

**Comment by diligent423** [comment]
Can we set the success_url like "example.dev://subscription/success" for mobile app?
